File: freeipa/src/main/java/com/sequenceiq/freeipa/service/registry/ServiceAddressResolver.java
package com.sequenceiq.freeipa.service.registry;

public interface ServiceAddressResolver {
    String resolveUrl(String serverUrl, String protocol, String serviceId) throws ServiceAddressResolvingException;

    String resolveHostPort(String host, String port, String serviceId) throws ServiceAddressResolvingException;
}

File: autoscale/src/main/java/com/sequenceiq/periscope/service/registry/ServiceAddressResolver.kt
package com.sequenceiq.periscope.service.registry

interface ServiceAddressResolver {
    @Throws(ServiceAddressResolvingException::class)
    fun resolveUrl(serverUrl: String, protocol: String, serviceId: String): String

    @Throws(ServiceAddressResolvingException::class)
    fun resolveHostPort(host: String, port: String, serviceId: String): String
}
